zeebe/bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/impl/instance/zeebe/ZeebeFormDefinitionImpl.java

```diff
@@ -90,6 +102,12 @@ public static void registerType(final ModelBuilder modelBuilder) {
             .namespace(BpmnModelConstants.ZEEBE_NS)
             .build();
 
+    bindingTypeAttribute =
+        typeBuilder
+            .enumAttribute(ZeebeConstants.ATTRIBUTE_BINDING_TYPE, ZeebeBindingType.class)
+            .namespace(BpmnModelConstants.ZEEBE_NS)
+            .build();
```

Comment:
I decided not to set a default value here, because the binding type is not applicable when using `formKey` or `externalReference`.

Reply:
Not setting a default means that existing models with a FormDefinition using `formId` (those without `bindingType`) are implicitly using `latest` even though there is no `bindingType` specified and no default either. I'm not sure whether it's a big problem, but perhaps we see a better solution. 

❓ Do you see any way to overcome this?

Reply:
Thanks for raising this 👍🏻.

> Not setting a default means that existing models with a FormDefinition using `formId` (those without `bindingType`) are implicitly using `latest` even though there is no `bindingType` specified and no default either.

Hm, I think the problem is rather that for these models the `bindingType` will be `null` and not `latest`, right? However, `latest` would make sense as this is already the current behavior in Zeebe if a `formId` is used.

I would basically see three options to deal with this:
1. Always use `latest` as the default if the `bindingType` is not explicitly set in the model, even if `formKey` or `externalReference` is used. This would be the simplest solution (and also resolve your [other comment](https://github.com/camunda/camunda/pull/19766#discussion_r1658848845)).

2. Use `latest` as the default if the `bindingType` is not explicitly set in the model, but _only_ if the `formId` is set (probably by enhancing the `getBindingType` method). This would add a bit of extra complexity, but reflect my [original intention](https://github.com/camunda/camunda/pull/19766#discussion_r1655076741).

3. Keep the current implementation. This would require an extra `null` check in Zeebe later when the binding type is actually used to resolve the form by `formId` (`null` would be equivalent to `latest` then).

I would prefer 1 or 2, also to be in line with `ZeebeCalledElementImpl` and `ZeebeCalledDecisionImpl` which actually define a default value.

Reply:
It seems to me that semantically it's not incorrect to use `latest` with `formKey`. There simply is only one version of it. 🤷 

I'm not a fan of unnecessary complexity. It doesn't appear to me as something that significantly helps our users. If it does, we could always add such rules later as improvements.

Given that option 1 is the simplest, it has my preference until we see a user need.

Reply:
Let's go with option 1 then, thanks! :)